eng/packages/http-client-csharp/generator/Azure.Generator/src/AzureInputLibrary.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Generator.CSharp.Input;
+using System;
+using System.Linq;
+
+namespace Azure.Generator
+{
+    /// <inheritdoc/>
+    public class AzureInputLibrary : InputLibrary
+    {
+        /// <inheritdoc/>
+        public AzureInputLibrary(string codeModelPath) : base(codeModelPath)
+        {
+        }
+
+        /// <summary>
+        /// Identify if the input is generated for Azure ARM.
+        /// </summary>
+        internal Lazy<bool> IsAzureArm => new Lazy<bool>(() => InputNamespace.Clients.Any(c => c.Decorators.Any(d => d.Name.Equals("Azure.ResourceManager.@armProviderNamespace"))));
```

Comment:
How will this be used? Couldn't we just add this property to the AzureClientPlugin and thereby avoid introducing this new type?

Reply:
We can, but I think it fits more in `InputLibray` than `Plugin`, if we keep adding random properties in to `Plugin`, it gets less structured.

Reply:
We aren't really adding any new functionality about the InputLibrary so an inheritance relationship doesn't really fit - at least not at this point. We are instead just calculating whether a specific decorator is found on the input library. This could be computed once in the AzureClientPlugin constructor.

Reply:
As offline discussed, we will hold the inheritance of `InputLibrary` until we need new functionality of input. 
Updated.